polynomial/__init__.py

```diff
@@ -203,16 +204,23 @@ def __bool__(self):
 
     def __add__(self, other):
         """Return self + other."""
+        if not isinstance(other, Polynomial):
+            if isinstance(other, (int, float, complex)):
```

Comment:
I am thinking of whether we should check for the same in `__mul__` and `__sub__` as well? If you agree, the `(int, float, complex)` tuple can be reused as a constant property in the class (or maybe as a global constant in the file since the decorator uses it as well) because later on more types might be added to the list, such as `Fraction`s, etc.

Reply:
I decided to implement this as a decorator which casts other if needed, and raises an error if other is not a numerical type.